components/script/dom/utils.rs

```diff
@@ -0,0 +1,83 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+use dom::bindings::codegen::Bindings::HTMLCollectionBinding::HTMLCollectionMethods;
+use dom::bindings::codegen::Bindings::NodeBinding::NodeMethods;
+use dom::bindings::codegen::InheritTypes::ElementCast;
+use dom::bindings::codegen::InheritTypes::HTMLElementBase;
+use dom::bindings::codegen::InheritTypes::HTMLElementCast;
+use dom::bindings::codegen::InheritTypes::NodeBase;
+use dom::bindings::codegen::InheritTypes::NodeCast;
+use dom::bindings::error::Error;
+use dom::bindings::error::{ErrorResult, Fallible};
+use dom::bindings::js::{Root, RootedReference};
+use dom::bindings::utils::Reflectable;
+use dom::element::Element;
+use dom::htmlcollection::HTMLCollection;
+use dom::htmlelement::HTMLElement;
+use dom::node::Node;
+use std::iter;
+
```

Comment:
So I'm not a *huge* fan with how these generic functions turned out. If anyone has any suggestions, let me know